desktop/libs/aws/src/aws/s3/s3fs.py

```diff
@@ -478,6 +478,9 @@ def _copy(self, src, dst, recursive, use_src_basename):
       if not src_key.endswith('/'):
         cut += 1
 
+    # Initially set to False assuming copy operation will work fine.
+    skip_copy = False
```

Comment:
Why not check at the top of this function to see if src and dst are the same and if so cancel the operation?

Reply:
Good point. I've updated my fix with this approach.